### PR TITLE
Fix test pollution by tearing down components properly

### PR DESCRIPTION
**Pull request by cyril-sf**

### CODE REVIEW

File: tests/unit/multi_select_test.coffee [hunk 1]
```diff
@@ -5,39 +7,43 @@ moduleForComponent 'multi-select', '[Unit] Multi select component',
     'template:select-item-layout'
     'template:select-item'
   ]
+  teardown: ->
+    Ember.run ->
+      multiSelect.destroy()
+    multiSelect = null
```
Comment (embooglement) [inline tests/unit/multi_select_test.coffee]: How does this work? Wouldn't `multiSelect` be `null` when the run loop is called?

**Comment by embooglement** [comment]
How exactly does this fix things?

File: tests/unit/multi_select_test.coffee (hunk 1 above)
Comment (jordansmith42) [inline tests/unit/multi_select_test.coffee]: Yeah I agree this seems strange that it wouldn't be inside the ember.run

Comment (cyril-sf) [inline tests/unit/multi_select_test.coffee]: `Ember.run` doesn't schedule anything, it executes the function inside a run loop.

So the execution order isn't changed.

Comment (embooglement) [inline tests/unit/multi_select_test.coffee]: Ah, I see. This seems fine then.

**Comment by embooglement** [comment]
LGTM

**Comment by cyril-sf** [comment]
The component is not torn down properly and an event handler remains set. Later, a test triggers the event, but the component doesn't exist anymore.

This makes sure that each instantiated components is torn down, which will remove the event handler.

**Pull request merged**